### PR TITLE
Its python cheatsheet

### PR DESCRIPTION
**Pull request by IamRafy**
--------------------------------------------------------------------
IA Page: https://duck.co/ia/view/python_cheat_sheet

### CODE REVIEW

**Comment by MariagraziaAlastra** [comment]
Hey @IamRafy, thanks for your contribution!
Could you please fill in the [PR template for CheatSheets](https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/pull_request_template_cheat_sheet.md) in your first comment? Thank you!

**Comment by mintsoft** [comment]
@MariagraziaAlastra wow you got to that quickly! Beat me to it

File: share/goodie/cheat_sheets/json/blender.json [hunk 1]
```diff
@@ -28,78 +28,78 @@
...
+            "val": "args",
+            "key": "Passed to script or command"
         }, {
-            "val": "Clear parent",
-            "key": "[Alt] [p]"
+            "val": "",
+            "key": "if no scriptFile or command ,Python enters interactive mode "
         }, {
             "val": "Track to",
             "key": "[Ctrl] [t]"
```
Comment (bsstoner) [inline share/goodie/cheat_sheets/json/blender.json]: Might want to revert the changes to this file. Did you intend to change the `blender.json` cheatsheet too?

Comment (IamRafy) [inline share/goodie/cheat_sheets/json/blender.json]: Okay on it.  Work in progress.

Regards :)
Mohammed Rafy

Comment (bsstoner) [inline share/goodie/cheat_sheets/json/blender.json]: Awesome :+1:

File: share/goodie/cheat_sheets/json/python.js [hunk 2]
```diff
@@ -0,0 +1,558 @@
...
+            "key": "print a help message and exit"
+        }, {
+            "val": "-i",
+            "key": "Inspect interactively after runnning script and force prompts even if stdin appears not to be a terminal"
+        }, {
+            "val": "-m module",
+            "key": "Search for module on sys.path and runs the module as a script"
+        }, {
+            "val": "-O",
+            "key": "Optimized generated bytecode ,Asserts are suppressed"
```
Comment (jagtalon) [inline share/goodie/cheat_sheets/json/python.js]: Should the comma be a `.` instead?

File: share/goodie/cheat_sheets/json/python.js [hunk 3]
```diff
@@ -0,0 +1,558 @@
...
+            "key": "Issues warnings on inconsistent tab usage"
+        }, {
+            "val": "-u",
+            "key": "unbuffered binary stdout and stderr"
+        }, {
+            "val": "-U",
+            "key": "Force python to interpret all strings literals as unicode literals"
+        }, {
+            "val": "-v",
+            "key": "Verbose(trace import statements)"
```
Comment (jagtalon) [inline share/goodie/cheat_sheets/json/python.js]: Add a space after "Verbose"

File: share/goodie/cheat_sheets/json/python.js [hunk 4]
```diff
@@ -0,0 +1,558 @@
...
+            "key": "Don't perform import-site on initialization"
+        }, {
+            "val": "-t",
+            "key": "Issues warnings on inconsistent tab usage"
+        }, {
+            "val": "-u",
+            "key": "unbuffered binary stdout and stderr"
+        }, {
+            "val": "-U",
+            "key": "Force python to interpret all strings literals as unicode literals"
```
Comment (jagtalon) [inline share/goodie/cheat_sheets/json/python.js]: Capitalize "python"

File: share/goodie/cheat_sheets/json/python.js [hunk 5]
```diff
@@ -0,0 +1,558 @@
...
+        }],
+        "General": [{
+            "val": "-W arg",
+            "key": "Warning control"
+        }, {
+            "val": "-x",
+            "key": "Skip first line of source ,allowing use of non-unix forms of #!cmd"
+        }, {
+            "val": "-3",
+            "key": "emit a DeprecationWarning for Python 3.x incompatibilities "
```
Comment (jagtalon) [inline share/goodie/cheat_sheets/json/python.js]: Capitalize "emit"

File: share/goodie/cheat_sheets/json/python.js [hunk 6]
```diff
@@ -0,0 +1,558 @@
...
+            "val": "scriptFile",
+            "key": "The name of a python file (.py) to execute. Read from stdin. "
+        }, {
+            "val": "-",
+            "key": "program read from stdin"
+        }, {
+            "val": "args",
+            "key": "Passed to script or command"
+        }, {
+            "val": "",
```
Comment (jagtalon) [inline share/goodie/cheat_sheets/json/python.js]: No value?

**Comment by moollaza** [comment]
@IamRafy sorry for the delay! Please leave us a comment after you make updates in the future so we can see that you've made a change.

It looks like you've accidentally combined the Blender Cheat Sheet with your Python Cheat Sheet? You have sections like "Navigation", "Animation" etc. Can you please remove everything that doesn't belong and leave a comment once the changes are made? Apologies for the delays and difficulties!

**Comment by IamRafy** [comment]
I have changed the blender Json file a long ago.  Anyhow,  I haven't
commented in idea page of python sheet related to the changes I make.  Is
that correct?

**Comment by moollaza** [comment]
>I have changed the blender Json file a long ago. 

@IamRafy it looks like you've somehow combined these changes into your Python Cheat Sheet. Please take a look at it and confirm that everything is correct.

>Anyhow, I haven't commented in idea page of python sheet related to the changes I make. Is that correct?

Sorry, I'm not quite sure what you mean by this. You don't need to comment on the Idea Page, if that's what you're asking.

I just need you to go over you cheat sheet here: https://github.com/duckduckgo/zeroclickinfo-goodies/pull/1456/files and make sure that it's all correct. Right now I'm seeing a bunch of content from the Blender Cheat Sheet, which obviously isn't related to Python Syntax.

**Comment by moollaza** [comment]
@IamRafy Ping! I'm just checking in to see if you're still interested in finishing this one up? We'd love to see this go live!

There's absolutely no rush, just wanted to make sure you're not stuck on anything.

**Comment by IamRafy** [comment]
Yes.  It's in progress.  It's a long one.  I will complete this one.

**Comment by moollaza** [comment]
@IamRafy I think it would be best to cover just the basics and essentials for now. Take a look at the other Language Cheat Sheets we have (e.g. Lua) which are more concise. The cheat sheet shouldn't really be lengthy and exhaustive, otherwise it will become too much to look at, and hard to navigate.

**Comment by IamRafy** [comment]
Got it .

**Comment by IamRafy** [comment]
Its done.. The cheat sheet . Check out my PR .

On Tue, Sep 29, 2015 at 11:28 PM, Zaahir Moolla <notifications@github.com>
wrote:

> @IamRafy <https://github.com/IamRafy> I think it would be best to cover
> just the basics and essentials for now. Take a look at the other Language
> Cheat Sheets we have (e.g. Lua) which are more concise. The cheat sheet
> shouldn't really be lengthy and exhaustive, otherwise it will become too
> much to look at, and hard to navigate.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/duckduckgo/zeroclickinfo-goodies/pull/1456#issuecomment-144136209>
> .
>



-- 
Best Regards ,
Twitter <https://twitter.com/RafyAsarmatta>  | Facebook
<https://www.facebook.com/asarmatta>
* ideabloom.wordpress.com *
+91-7204945348

**Comment by IamRafy** [comment]
Is it all correct Zac ?

On Thu, Oct 1, 2015 at 10:48 PM, Zac Pappis <notifications@github.com>
wrote:

> Closed #1456
> <https://github.com/duckduckgo/zeroclickinfo-goodies/pull/1456>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/duckduckgo/zeroclickinfo-goodies/pull/1456#event-424469804>
> .
>



-- 
Best Regards ,
Twitter <https://twitter.com/RafyAsarmatta>  | Facebook
<https://www.facebook.com/asarmatta>
* ideabloom.wordpress.com *
+91-7204945348

**Comment by moollaza** [comment]
@IamRafy We closed this because it looks like you continued it in #1622 ?

**Comment by moollaza** [comment]
Both of these look the same though, but this PR has broken JSON syntax...

**Comment by IamRafy** [comment]
So simply I wasted my time .. ! It was a new file . it was not a
blender.Json.

**Comment by moollaza** [comment]
@IamRafy sorry about the confusion! You certainly haven't wasted your time. I've responded to you on Slack, explaining what we thought was going on. We can definitely move forward with either PR, it seems like we don't need *both* though?
